[AIRFLOW-3905] Allow using "parameters" in SqlSensor

XD-DENG opened the pull request:
### Jira

  - https://issues.apache.org/jira/browse/AIRFLOW-3905

### Description

In most SQL-related operators/sensors, argument `parameters` is available to help render SQL command conveniently, like `MySqlOperator` (https://github.com/apache/airflow/blob/master/airflow/operators/mysql_operator.py#L25) and `PostgresOperator` (https://github.com/apache/airflow/blob/master/airflow/operators/postgres_operator.py#L24).

But this is not available in `SqlSensor` yet.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Tests passed.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

XD-DENG committed:
[AIRFLOW-3905] Allow 'parameters' in SqlSensor

zhongjiajie commented on airflow/sensors/sql_sensor.py:
```diff
@@ -34,22 +34,25 @@ class SqlSensor(BaseSensorOperator):
-        self.log.info('Poking: %s', self.sql)
-        records = hook.get_records(self.sql)
+        self.log.info('Poking: %s (with parameters %s)', self.sql, self.parameters)
+        records = hook.get_records(self.sql, self.parameters)
```
Should we check if `self.parameters` is None in `get_records` function?
```suggestion
        if self.parameters:
            records = hook.get_records(self.sql, self.parameters)
        else:
            records = hook.get_records(self.sql)
```

XD-DENG replied:
Hi @zhongjiajie, It's not necessary to do this check here, as it's already handled inside `hook.get_records`.

You can refer to https://github.com/apache/airflow/blob/master/airflow/hooks/dbapi_hook.py#L98

zhongjiajie replied:
@XD-DENG Ok, I get it, thinks

zhongjiajie reviewed (commented):
Please find my two cents.

feng-tao commented on airflow/sensors/sql_sensor.py:
```diff
@@ -34,22 +34,25 @@ class SqlSensor(BaseSensorOperator):
-        self.log.info('Poking: %s', self.sql)
-        records = hook.get_records(self.sql)
+        self.log.info('Poking: %s (with parameters %s)', self.sql, self.parameters)
+        records = hook.get_records(self.sql, self.parameters)
```
Do all the sql type hooks(druid, mysql, postgres etc) are all inherited from dbapi hook? If that's case, the change LGTM

feng-tao replied:
If not, we should change that hook to inherit from dbapi hook.

XD-DENG replied:
Yes, hooks of druid, mysql, mssql, Postgres, oracle etc are all inherited from dbapi hook.

reference:
- https://github.com/apache/airflow/blob/master/airflow/hooks/mssql_hook.py#L25
- https://github.com/apache/airflow/blob/master/airflow/hooks/mysql_hook.py#L28
- https://github.com/apache/airflow/blob/master/airflow/hooks/oracle_hook.py#L29
- https://github.com/apache/airflow/blob/master/airflow/hooks/postgres_hook.py#L28

For `SqlSensor`, the `get_hook()` in `BaseHook.get_connection(self.conn_id).get_hook()` will decide which exact hook to use based on the connection type (reference: https://github.com/apache/airflow/blob/master/airflow/models/connection.py#L184).

feng-tao replied:
But I check`CloudSqlDatabaseHook `(https://github.com/apache/airflow/blob/6b38649fa6cdf16055c7f5458050c70f39cac8fd/airflow/contrib/hooks/gcp_sql_hook.py#L669) which doesn't seem to inherit from dbapi hook. 

But the I assume the sqlSensor could use CloudSqlDatabasehook? Given there are many connection hooks defined, a safer approach would be check if the hook is an instance of dbapi hook, use the `records = hook.get_records(self.sql, self.parameters)`, otherwise fall back to `records = hook.get_records(self.sql)`. what you think?

XD-DENG replied:
Actually `SqlSensor` can not support `CloudSqlDatabaseHook`. 

Eventually `SqlSensor` uses `get_records` method to retrieve the records from DB, while `get_records` method is not available in `CloudSqlDatabaseHook` at all.

feng-tao replied:
I see, but would a check on hook type be safer?

XD-DENG replied:
Get what you mean.

Given the limitation in https://github.com/apache/airflow/blob/master/airflow/models/connection.py#L184 and the implementation of each hook, only the connection types below are supported by `SqlSensor`:

- 'google_cloud_platform'
- 'jdbc'
- 'mssql'
- 'mysql'
- 'oracle'
- 'postgres'
- 'presto'
- 'sqlite'
- 'vertica'

I will add a check.

XD-DENG committed:
Add check on conn_type & add test

Not all SQL-related connections are supported by SqlSensor,
due to limitation in Connection model and hook implementation.
